content/guide/core/file-system.md

```diff
@@ -0,0 +1,747 @@
+```
+
+## FileSystem API
+
+### documents()
+
+```ts
+const folder: Folder = knownFolders.documents()
+```
+
+Gets the Documents folder available for the current application. This Folder is private for the application and not accessible from Users/External apps.
+
+---
+
+### externalDocuments()
+
+```ts
+const folder: Folder = knownFolders.externalDocuments()
+```
+
+Gets the Documents folder available for the current application on an external storage. This Folder is private for the application and not accessible from Users/External apps. On android this requires `READ_EXTERNAL_STORAGE/WRITE_EXTERNAL_STORAGE` permissions. There is no external storage on iOS os it is the same as [documents()](#documents).
+
+---  
+
+### temp()
+
+```ts
+const folder: Folder = knownFolders.temp()
+```
+Gets the Temporary (Caches) folder available for the current application. This Folder is private for the application and not accessible from Users/External apps.
+
+---
+
+### currentApp()
+
+```ts
+folder: Folder = knownFolders.currentApp()
+```
+
+Gets the root folder for the current application.
+
+---
+
+### iOS known folders
+
+The following methods allow to access to iOS known folders.
+
+#### library()
+```ts
+folder: Folder = knownFolders.ios.library()
+```
+Gets the NSLibraryDirectory.
+
+---
+
+#### developer()
+```ts
+folder: Folder = knownFolders.ios.developer()
+```
+Gets the NSDeveloperDirectory.
+
+---
+
+#### desktop()
+```ts
+folder: Folder = knownFolders.ios.desktop()
+```
+Gets the NSDesktopDirectory.
+
+---
+
+#### downloads()
+```ts
+folder: Folder = knownFolders.ios.downloads()
+```
+Gets the NSDownloadsDirectory.
+
+---
+
+#### movies()
+```ts
+folder: Folder = knownFolders.ios.movies()
+```
+Gets the NSMoviesDirectory.
+
+---
+
+#### music()
+```ts
+folder: Folder = knownFolders.ios.music()
+```
+Gets the NSMusicDirectory.
+
+---
+
+#### pictures()
+```ts
+folder: Folder = knownFolders.ios.pictures()
+```
+Gets the NSPicturesDirectory.
+
+---
+
+#### sharedPublic()
+```ts
+folder: Folder = knownFolders.ios.sharedPublic()
+```
+Gets the NSSharedPublicDirectory.
+
+---
+
+### fromPath()
+
+```ts
+
+file: File = File.fromPath(path)
+
+```
+
+or 
+
+```ts
+folder: Folder = Folder.fromPath(path)
+```
+
+Gets or creates a Folder or File entity at the specified path.
+
+---
+
+### getFolder
+```ts
+folder : Folder = folder.getFolder(name)
+```
+
+Gets or creates a Folder entity with the specified `name` within a Folder.
+
+---
+
+
+### exists
+
+```ts
+folderExists: boolean = Folder.exists(path)
+```
+
+or 
+
+```ts
+file: boolean = File.exists(path)
+```
+
+Checks whether a Folder or File with the specified path already exists.
+
+---
+
+### isKnown
+
+```ts
+isItAKnownFolder: boolean = folder.isKnown
+```
+Determines whether this instance is a known folder (accessed through the `knownFolders` object).
+
+---
+Both the File and Folder classes extend the FileSystemEntity which has the following API:
```

Comment:
Both the File and Folder classes extend the FileSystemEntity, which has the following API: